pkgs/by-name/vo/volk/package.nix

```diff
@@ -0,0 +1,47 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  fetchpatch,
+  cmake,
+  python3,
+  enableModTool ? true,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "volk";
+  version = "3.1.2";
+
+  src = fetchFromGitHub {
+    owner = "gnuradio";
+    repo = "volk";
+    rev = "v${finalAttrs.version}";
```

Comment:
Should this be `"rev = v${version}";`

I realize it's probably copypasta from previous where versions were different, but is this normal in other cases?

Reply:
> Should this be `rev = "v${version}";`
> 
> I realize it's probably copypasta from previous where versions were different, but is this normal in other cases?

See:

https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes

Reply:
I'll take your word for it.

I can't see why this needs `finalAttrs` or to be a function rather than plain data (which I think are the options here)

Reply:
> I'll take your word for it.
> 
> I can't see why this needs `finalAttrs` or to be a function rather than plain data (which I think are the options here)

Now I see that the docs don't motivate people to switch to `finalAttrs`, which is a shame... In general the keypoints to do that are:

1. Easier overriding
2. Not having to use `rec` which is discouraged and is evaluation expensive (see [this](https://nix.dev/guides/best-practices#recursive-attribute-set-rec)).
3. Nicer expressions if you need to reference `finalAttrs.finalPackage`.

Reply:
In general sure, all good points.

None apply here?

(I have zero experience with nix maintenance)

Reply:
> In general sure, all good points.
> 
> None apply here?

Why not? Only point number 3 doesn't apply.